File: introducao-a-computacao/2022/trabalho-pratico/python/domain/contador.py
from .resultado import ResultadoContagem
from typing import List


def contar_pares_impares(numeros: List[int]) -> ResultadoContagem:
    pares = sum(1 for n in numeros if n % 2 == 0)
    impares = len(numeros) - pares
    return ResultadoContagem(pares=pares, impares=impares)


def formatar_resultado(r: ResultadoContagem) -> str:
    return (
        "\n===== Resultado =====\n"
        f"Quantidade de números pares: {r.pares}\n"
        f"Quantidade de números ímpares: {r.impares}"
    )

File: introducao-a-computacao/2022/trabalho-pratico/c/domain/contador.c
#include "contador.h"

ResultadoContagem contar_pares_impares(const int *numeros, int n)
{
    ResultadoContagem r = {0, 0};
    if (numeros == NULL || n <= 0)
        return r;

    for (int i = 0; i < n; i++) {
        if (numeros[i] % 2 == 0)
            r.pares++;
        else
            r.impares++;
    }
    return r;
}
